### ISSUE TITLE
Make it compatible with Spring Boot 3.0.0

### ISSUE THREAD

**Issue by mmaryo** [DataDog/dd-trace-java#4477]
Hello 
I spoke with the support and he told me to open a feature request for Spring Boot 3

```
I discussed this with our internal team and we found that support for Java 19 is currently in Beta and Spring 3.0.0 and 6 are currently unsupported at this time. For Beta,  support for new features and bug and security fixes are provided on a best-effort basis.
 ```

**Comment by jaredtbates** [comment]
Hello, we're running into issues with Spring Boot 3 auto-instrumentation, namely we are not seeing path parameters in our APM traces (instead, we see `?`), and we are not seeing errors/exceptions that are thrown. Can we get an ETA on when Spring Boot 3 support will be available?

**Comment by sebphil** [comment]
Hi! I'm having an issue which seems very similar to this one. I upgraded my application to Spring Boot 3, and I can no longer see the Spring WebFlux / WebClient spans in the APM traces (http.request and http.client.request) that I had on Spring Boot 2.7. I can only see Netty spans (which were also there with Spring Boot 2.7, but are of no use to me). I tried instrumenting my application with the latest version of dd-java-agent (1.7.0). According to the release notes, version 1.7.0 should support Spring 6 / Spring Boot 3, but the issue is still there and the Spring WebFlux / WebClient spans are still missing.

**Comment by mcculls** [comment]
#4605 added support for Spring Boot 3 and Spring Web (MVC) 6 but I believe further work is needed to fully support WebFlux 6 - @amarziali can you confirm?

**Comment by amarziali** [comment]
> #4605 added support for Spring Boot 3 and Spring Web (MVC) 6 but I believe further work is needed to fully support WebFlux 6 - @amarziali can you confirm?

Looks like support for web client is missing for spring 6. Servers (webmvc and webflux) should be ok

**Comment by sebphil** [comment]
I'm not so sure about that. According to Spring docs, server requests are instrumented and should produce spans labeled "http.server.request" (cf https://docs.spring.io/spring-framework/docs/current/reference/html/integration.html#integration.observability.http-server), but I'm not seeing those spans in the traces in APM.

**Comment by devinsba** [comment]
We have our own naming conventions for observability names. The top level spans for webflux should be `netty.request` and the top level for webmvc will be `<server>.request` depending on what server you are using, ex: tomcat, undertow, etc.

**Comment by jleemoneylion** [comment]
Hi! Looks like JMS instrumentation is also broken in Spring Boot 3+. I suspect it is due to the migration from `javax` to `jakarta` package name introduced by Spring Boot 3+.
https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/instrumentation/jms/src/main/java/datadog/trace/instrumentation/jms/JMSMessageProducerInstrumentation.java#L26
https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/instrumentation/jms/src/main/java/datadog/trace/instrumentation/jms/JMSMessageConsumerInstrumentation.java#L30

**Comment by bm1549** [comment]
This was fixed via https://github.com/DataDog/dd-trace-java/pull/5868 and released as part of [v1.21.0](https://github.com/DataDog/dd-trace-java/releases/tag/v1.21.0). The latest version of the tracer is [v1.24.2](https://github.com/DataDog/dd-trace-java/releases/tag/v1.24.2)

If the issue persists, please comment here and we can re-open the issue
